File: backend/app/schemas/system.py
from pydantic import BaseModel


class PhddnsWatchdogStatusResponse(BaseModel):
    installed: bool
    enabled: bool
    running: bool
    pid: int | None = None
    last_exit_status: int | None = None
    label: str
    plist_path: str
    log_lines: list[str] = []
    message: str = ""


class PhddnsWatchdogTogglePayload(BaseModel):
    enabled: bool

File: frontend/src/types/system.ts
export interface PhddnsWatchdogStatus {
  installed: boolean
  enabled: boolean
  running: boolean
  pid: number | null
  last_exit_status: number | null
  label: string
  plist_path: string
  log_lines: string[]
  message: string
}

export interface PhddnsWatchdogTogglePayload {
  enabled: boolean
}
